org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/WizardNewXtextProjectCreationPage.java

```diff
@@ -228,6 +229,19 @@ protected boolean validatePage() {
 		return true;
 	}
 
+	private IStatus validateProjectName() {
+		String projectName = getProjectName();
+		IStatus status = JavaConventions.validatePackageName(projectName, JavaCore.VERSION_1_5, JavaCore.VERSION_1_5);
+		if (status.isOK()) {
+			// https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.8
+			// for historical reasons only underscore and $ are accepted as package name, but not for Xtext projects  
+			if (projectName.contains("$")) {
+				status = Status.CANCEL_STATUS;
```

Comment:
Just a minor thing to be consistent: an ERROR status should be created here like it is done in `org.eclipse.jdt.core.JavaConventions#validatePackageName(String, String, String)`